self_enrollment_tool/views.py

```diff
@@ -0,0 +1,267 @@
+
+from .utils import get_canvas_roles, install_unenrollment_tool
+
+logger = logging.getLogger(__name__)
+
+SDK_CONTEXT = SessionInactivityExpirationRC(**settings.CANVAS_SDK_SETTINGS)
+
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_SELF_ENROLLMENT_TOOL)
+@require_http_methods(['GET'])
+def index(request):
+    try:
+        # The school that this tool is being launched in
+        tool_launch_school = request.LTI['custom_canvas_account_sis_id'].split(':')[1]
+    except Exception:
+        logger.exception('Error getting launch school')
+
+    # Self-enroll course instance IDs
+    course_instance_ids = {
+        crs_id[0] for crs_id in SelfEnrollmentCourse.objects.values_list('course_instance_id')}
+
+    # Additional course info for the tool launch school courses
+    course_info = CourseInstance.objects.filter(
+        course_instance_id__in=course_instance_ids, course__school=tool_launch_school)
+
+    # Tool launch school course instance IDs
+    course_instance_ids = {crs_id[0] for crs_id in course_info.values_list('course_instance_id')}
+
+    self_enroll_course_list = SelfEnrollmentCourse.objects.filter(
+        course_instance_id__in=course_instance_ids).order_by('-last_updated')
+
+    # Update "updated_by" ids to first and last name and add other
+    # relevant data from course_info_updater (CourseInstance objects), etc..
+    updater_ids = {course.updated_by for course in  self_enroll_course_list}
+    updaters = SimplePerson.objects.get_list_as_dict(user_ids=updater_ids)
+    user_roles = UserRole.objects.all()
+
+    for course in self_enroll_course_list:
+        updater = updaters.get(course.updated_by)
+        if updater:
+            course.last_modified_by_full_name = f'{updater.name_first} {updater.name_last}'
+
+        try:
+            course_info_updater = course_info.get(course_instance_id=course.course_instance_id)
+            if course_info_updater:
+                course.course = course_info_updater.course
+                course.section = course_info_updater.section
+                course.term = course_info_updater.term
+                course.title = course_info_updater.title
+                course.short_title = course_info_updater.short_title
+                course.sub_title = course_info_updater.sub_title
+        except CourseInstance.DoesNotExist:
+            logger.exception(f'This course instance ID {course.course_instance_id} '
+                             f'does no exist in course instance database table')
+
+        if course.role_id:
+            course.role_name = user_roles.get(role_id=course.role_id).role_name
+
+        course.self_enroll_url = f'https://{settings.SELF_ENROLL_HOSTNAME}/{course.uuid}'
+
+    context = {
+        'self_enroll_course_list': self_enroll_course_list
+    }
+    return render(request, 'self_enrollment_tool/index.html', context=context)
+
+
+@login_required
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Your code allows safe HTTP methods (i.e `GET`, `HEAD`, `OPTIONS`) and unsafe HTTP methods (i.e `POST`, `PUT`, `DELETE`) to perform an operation on a web application. This could impact its security.  We recommend that you to use safe http methods while performing read-only operations. If you have to use unsafe HTTP methods, use them only to change the state of an appilication. [Learn more](https://cwe.mitre.org/data/definitions/352.html).